libs/langgraph/langgraph/pregel/__init__.py

```diff
@@ -1176,6 +1216,9 @@ def put_checkpoint(metadata: CheckpointMetadata) -> Iterator[Any]:
 
                 # set final channel values as run output
                 run_manager.on_chain_end(read_channels(channels, output_keys))
+        except GraphInterrupt as e:
+            run_manager.on_chain_end(e)
```

Comment:
@nfcampos i don't think this is actually working, i still see error getting logged in langsmith (even though we are catching the interrupt correctly here). am i doing anything wrong?

<img width="825" alt="image" src="https://github.com/user-attachments/assets/6591938b-9133-46e8-ae4e-bc69508cd6ef">

Reply:
I think passing an error to on_chain_end is wrong?

Reply:
yea, fair enough -- what should we pass instead?
i'll update that, but the issue is unrelated

Reply:
pass `read_channels(channels, output_keys)` as we do in the line above?

Reply:
but we don't have access to `ChannelsManager` at that point, should we add it there?